firebase-firestore/src/commonMain/kotlin/dev/gitlive/firebase/firestore/FieldValueSerializer.kt

```diff
@@ -0,0 +1,29 @@
+package dev.gitlive.firebase.firestore
+
+import dev.gitlive.firebase.FirebaseDecoder
+import dev.gitlive.firebase.FirebaseEncoder
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.descriptors.buildClassSerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+
+/** A serializer for [FieldValue]. Must be used in conjunction with [FirebaseEncoder]. */
+object FieldValueSerializer : KSerializer<FieldValue> {
+    override val descriptor = buildClassSerialDescriptor("FieldValue") { }
+
+    override fun serialize(encoder: Encoder, value: FieldValue) {
+        if (encoder is FirebaseEncoder) {
+            encoder.value = value.platformValue
+        } else {
+            throw IllegalArgumentException("This serializer must be used with FirebaseEncoder")
```

Comment:
it'd be nice to serialize it, but `FieldValue.arrayUnion` and `arrayDelete` take `Any` and I have no idea how to serialize it in a clean way.. use Json maybe?..